libs/qlik/src/infoblock/vl-qlik-infoblock.component.ts

```diff
@@ -0,0 +1,119 @@
+import { css, CSSResult, html, PropertyDeclarations } from 'lit';
+import { BaseLitElement, registerWebComponents, webComponent } from '@domg-wc/common-utilities';
+import { vlElementsStyle } from '@domg-wc/elements';
+import { InfoblockVisualization } from '../utils/vl-qlik.interfaces';
+import { Qlik, STARDUST } from '@domg/qlik-lib';
+import { renderStack } from '../utils/qlik-render-utils';
+import { stardust } from '@nebula.js/stardust';
+import { VlInfoblockComponent, VlInfoTile, VlLoaderComponent } from '@domg-wc/components';
+import { VlQlikVisualComponent } from '../visual';
+
+@webComponent('vl-qlik-infoblock')
+export class VlQlikInfoblockComponent extends BaseLitElement {
+    title = '';
+    private icon = '';
+    visuals: InfoblockVisualization[][];
+    connection?: Qlik;
+    private stardust?: stardust.Embed;
+    private connected: boolean;
+
+    static {
+        registerWebComponents([VlInfoblockComponent, VlInfoTile, VlQlikVisualComponent, VlLoaderComponent]);
+    }
+
+    static get styles(): (CSSResult | CSSResult[])[] {
+        return [vlElementsStyle];
+    }
+
+    static get properties(): PropertyDeclarations {
+        return {
+            title: { type: String, attribute: 'title', reflect: true },
+            icon: { type: String, attribute: 'icon', reflect: true },
+            visuals: { type: Array },
+            connection: { type: Object },
+            connected: { type: Boolean, state: true },
+        };
+    }
+
+    async firstUpdated() {
+        if (this.connection && !this.connected) {
+            this.stardust = await STARDUST(this.connection.app);
+            this.connected = true;
+        }
+    }
+
+    render() {
+        if (!this.connected) {
+            return html` <vl-loader data-vl-text="Visualisatie is aan het laden"></vl-loader>`;
+        }
+        return html`
+            <style>
+                ${Array.from({ length: 13 }, (x, i) => i).flatMap((i) =>
```

Comment:
Is er een mogelijkheid om geen inline-styles te gebruiken?
Door inline-styles te gebruiken kan er geen strikte CSP-regel toegepast worden.